### DISCUSSION TITLE
Why should I care?

### DISCUSSION THREAD

**Discussion by Herringway** [dlang/PhobosV3-Design#31] [General]
This is an important question for a simple little nobody of a user like me with problems to solve. I want to be excited! This is the future of D, isn't it? But there needs to be reasons to want to use V3 over V2. I understand that these things take time and that it's still fairly early on in the process, but nothing in the design documents, discussions nor PRs so far is leading me to believe that V3 will have anything interesting to offer.

The only positive feature I've seen so far is the remove of autodecoding. I already try to work around that a lot, so the promise of not having to do that work anymore is welcome. But the rest of the design document suggests that in exchange for that, I will now have to do a lot more work to maintain the same level of functionality. No support for wchar or dchar? Trading Exceptions for return values? Changing the names of things (okay, sure, that bandaid should be ripped off at some point, but still...)? All this just sounds like it'll be work for me, and I just don't buy the justification. I don't care one bit about avoiding the GC. I already use D for bare metal game dev, and the only GC problem I have is that there ain't an implementation for bare metal.

The actual code written so far is just rearranging the furniture. Two months of development and all we have is part of std.traits? What year are we going to get algorithms? Or will we get there at all? At this glacial pace, I'm expecting people to get bored and abandon the project before anything interesting happens. Yes, I understand that these are building blocks for everything else, but what is "everything else"? Is is just the same stuff we have already, but with reduced functionality and old problems exchanged for new ones? Thrilling.

The only fruitful discussions so far have centred on renaming existing symbols and swapping paradigms. Any talk of new features or new interfaces is met with silence or "idk lol". Decades have passed since Phobos V2 was introduced. People with literal decades of experience with the design flaws of Phobos V2 are being ignored instead of supported. They're not going to get involved if they're simply brushed off, and those flaws will continue to persist in V3.

So, what is Phobos V3? Is it just rearranging the furniture? Or will it be a well-thought out reworking of the many portions of the library that desperately need some love?

Because I don't care where you put the worn-out sofa in the room.

**Comment by rikkimax** [comment]
Currently, people are working on the leaf symbols. Things that don't depend upon a huge amount of other things.

Which is a great place to begin.

Once we have sumtypes, I can get value type exceptions into queue. I don't know what is holding @WalterBright up on this. Although from what he said, he is converging towards more the feature set I propose for them, rather than what he has written up currently.

As for character types, I do not agree with that approach either.

But no, PhobosV3 isn't just a rearrangement, it's meant to also have things like eventloop and windowing. However, if I have anything to do with it, it'll be all in shared libraries. Right now it isn't, therefore not a me problem.

**Reply by Herringway** [reply]
Designs don't depend on symbols.  The folks who aren't working on those core leaf symbols could be working on designing the rest of Phobos V3, but it's a lot of commitment that I personally wouldn't be comfortable with if the only response is "meh" and a shrug. The eventloop and windowing do sound interesting, but I don't see any of that anywhere either; IIRC there's been promise of an std.eventloop for a pretty long time, too...

While I do believe you when you say you're working on things (I've always been very interested in your work) and I do believe that someday that work will make a positive impact on D as a language, I see no reason to believe that any of it will have any meaningful impact on Phobos V3 right now. 

You and I both know very well that D has a long history of being very discouraging for any contributors looking to shake things up in any significant way, and I don't wanna give up on the hope that it'll change someday. I'm the maintainer of D-YAML, but even though D-YAML has moved under the dlang-community umbrella, and it is now technically part of the core D infrastructure (dub), I still feel like I'm working on it entirely by myself. It's a popular "community-supported" package, but I can't get any support whatsoever, not even feedback on things, and that makes it difficult to find the motivation to do the work that needs to be done to update it to the latest standards and improve its efficiency. It really doesn't feel like we have a community at all most of the time, just a bunch of people throwing things at the wall to see if it sticks. It doesn't seem like we can work together at all.
